pages/instructorFileBrowser/instructorFileBrowser.js

```diff
@@ -207,54 +187,50 @@ function browseDirectory(file_browser, callback) {
         file_browser.dirs = [];
         async.eachOfSeries(
```

Comment:
@nwalters512 given we have some I/O in all of the cases I wonder if it makes sense to replace this `eachOfSeries` with a `Promise.all` for a `map`, so that they can run in parallel. Thoughts?

Reply:
I'm a little wary of `Promise.all` with an unbounded number of promises. If you want to add concurrency, I'd go with something like [`async.mapLimit`](https://caolan.github.io/async/v3/docs.html#mapLimit) with a low-ish concurrency limit (3?).